Blink/SceneDelegate.swift

```diff
@@ -308,3 +320,161 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
   @objc var spaceController: SpaceController { _spCtrl }
 
 }
+
+// MARK: Manage the `scene(_:openURLContexts:)` actions
+extension SceneDelegate {
+  
+  /**
+   Handles the `ssh://` URL schemes and x-callback-url for devices that are running iOS 13 or higher.
+   - Parameters:
+     - xCallbackUrl: The x-callback-url specified by the user
+   */
+  func handleSshUrlScheme(with sshUrl: URL) {
+    
+    var sshCommand = "ssh"
+    
+    // Progressively unwrap all of the parameters available on the URL to form
+    // the SSH command to be later passed to the shell
+    if let port = sshUrl.port {
+      sshCommand += " -p \(port)"
+    }
+    
+    if let username = sshUrl.user {
+      sshCommand += " \(username)@"
+    }
+    
+    if let host = sshUrl.host {
+      sshCommand += "\(host)"
+    }
+    
+    guard let term = _spCtrl.currentTerm() else {
+      return
+    }
+    
+    _spCtrl.focusOnShellAction()
+    
+    if term.isRunningCmd() {
+      // If a SSH/mosh connection is already open in the current terminal shell
+      // create a new one and then write the SSH command
+      
+      _spCtrl.newShellAction()
+      
+      guard let term = _spCtrl.currentTerm() else {
+        return
+      }
+      
+      DispatchQueue.main.asyncAfter(deadline: DispatchTime.now() + 0.5) {
+        term.termDevice.write(sshCommand)
+      }
+    } else {
+      // No running command or shell found running, run the SSH command on the
+      // available shell
+      term.termDevice.write(sshCommand)
+    }
+  }
+  
+  /**
+   Handles the x-callback-url, if  a successful `x-success` URL is provided when being called from apps like Shortcuts it returns to the original app after a successful execution.
+    - Parameters:
+      - xCallbackUrl: The x-callback-url specified by the user, URL format should be `blinkshell://run?key=KEY&cmd=CMD%20ENCODED`
+   */
+  func handleXcallbackUrl(with xCallbackUrl: URL) {
+    
+    guard let xCallbackUrlHost = xCallbackUrl.host, xCallbackUrlHost == "run" else {
```

Comment:
No need of xCallbackUrlHost if you don't use it elsewhere. Just use `xCallbackUrl.host == "run"`